raythurman2386/ravenwood-gallery
# pagination

# Add Pagination to Image Gallery

## Overview
This PR implements pagination for the image gallery to improve performance and user experience when dealing with a large number of images.

## Changes
- Implemented server-side pagination using Cloudinary's API
- Added pagination controls to the UI
- Updated `getServerSideProps` to fetch paginated results
- Modified the Home component to handle pagination state

## Implementation Details

### Server-side Changes
- Updated `getServerSideProps` to accept a `page` query parameter
- Implemented logic to fetch the correct set of images for each page using Cloudinary's `next_cursor`
- Added `totalImages`, `currentPage`, `imagesPerPage`, and `nextCursor` to the props returned by `getServerSideProps`

### Client-side Changes
- Added pagination controls (Previous/Next buttons) to the UI
- Implemented `handlePageChange` function to navigate between pages
- Updated the Home component to accept and use new pagination-related props

## How to Test
1. Navigate to the home page
2. Verify that only a subset of images (defined by `imagesPerPage`) is displayed
3. Use the pagination controls to navigate between pages
4. Confirm that different sets of images are loaded when changing pages
5. Verify that the "Previous" button is disabled on the first page
6. Verify that the "Next" button is disabled on the last page
7. Check that the current page number and total pages are displayed correctly

## Performance Considerations
- This change should significantly improve initial page load time and overall performance when dealing with a large number of images
- Server-side pagination ensures that we only fetch and process the necessary images for each page

## Future Improvements
- Implement client-side caching to improve navigation performance between already-visited pages
- Add option for users to change the number of images per page
- Implement infinite scrolling as an alternative to pagination buttons

## Related Issues
Closes #[Issue Number 3]
